infrastructure/cli/commands/redis/index.ts

```diff
@@ -46,32 +62,18 @@ export class Client {
 
   async connectToRedisLocally(
     port: Port,
-    callback: (redisClient: redis.RedisClient) => void | Promise<void>,
+    callback: (redisClient: RedisClientType) => void | Promise<void>,
   ): Promise<void> {
-    const portNum = typeof port === 'string' ? parseInt(port, 10) : port
-    process.stderr.write(`Connecting to redis on local port ${portNum}…\n`)
-    const redisClient = redis.createClient({
-      password: this.authToken,
-      host: 'localhost',
-      port: portNum,
-      tls: {
-        requestCert: false,
-        rejectUnauthorized: false,
-      },
-    })
+    const redisClient = createLocalRedisClient(port, this.authToken)
+
+    process.stderr.write(`Connecting to redis on local port ${port}…\n`)
+    await redisClient.connect()
     const value = callback(redisClient)
     if (value && 'then' in value) {
       await value
     }
```

Comment:
I may miss something here but I _think_ `await` is quite clever and could be used not only with Promises (usual case) but also to await non-promise values but also thenable (objects which have a `then()` method).

e.g. 

```javascript
const aValue = 'hello'
console.log(await aValue)  // hello

const emptiness = null
console.log(await emptiness)  // null

const promising = Promise.resolve(42)
console.log(await promising)  // 42

const thenable = {
    then() {
        console.log(`I'm a thenable`)
    }
}
console.log(await thenable)  // I'm a thenable
```

It may be that the `if (value && 'then' in value)` statement is not necessary?

Reply:
i've seen the `'then' in value` pattern in quite a few places, but perhaps it existed for non-await compatibility 🤔 
i'll look into changing _all_ the places i've used it.

Reply:
I **_think_** that's where it makes a difference, if you call `.then()` on a promise then that method needs to be there or it would blow-up for non-promises. But `await` seems to be a bit more easy going.

The `await` on a non-promise/value is so that a function could potentially return a promise or a result if that is already available (e.g. a local cache or something)

In this case, `callback()` could return nothing or it could return a promise. If it returns a promise this is waiting for the result...although admittedly I'm not 100% why we're awaiting for its result here.

If I understand this correctly, this `connectToRedisLocally()` function is a wrapper around a callback, it does something and then it calls `redisClient.quit()`.

Peraphs you can simplify this, remove the callback altogether and leave the closing to the caller, e.g.

```javascript
async function connectToRedisLocally(): Promise<RedisClient> {
  const redisClient = createLocalRedisClient(port, this.authToken)
  await redisClient.connect() 
  return redisClient
}

// client code

let redisClient = await connectToRedisLocally()

// do whatever needs to be done, e.g.
await clearSessions(redisClient)

process.stderr.write('Redis quitting…\n')
await redisClient.quit()
```

but then again, maybe that function could all go? 🤷

Reply:
having looked at all these `'then' in value` patterns it totally looks like i can just await even non-promises for cleaner code that effectively does the same thing. there might be the rare context switch when the value is not a promise but whatever.

>not 100% why we're awaiting for its result here
we deffo need to await the callback. if the callback is async (even one that resolves to void) and we don't await it, the next action immediately starts without the callback having finished. so redis is highly likely to quit too early